ingest/src/test/java/com/microsoft/azure/kusto/ingest/ResourceManagerTest.java

```diff
@@ -17,113 +20,108 @@
 
 class ResourceManagerTest {
 
-    private ResourceManager resourceManager;
-    private Client clientMock = mock(Client.class);
+    private static ResourceManager resourceManager;
+    private static Client clientMock = mock(Client.class);
 
     private static final String QUEUE_1 = "queue1";
     private static final String QUEUE_2 = "queue2";
     private static final String STORAGE_1 = "storage1";
     private static final String STORAGE_2 = "storage2";
     private static final String AUTH_TOKEN = "AuthenticationToken";
+    private static final String STATUS_TABLE = "statusTable";
+    private static final String FAILED_QUEUE = "failedQueue";
+    private static final String SUCCESS_QUEUE = "successQueue";
 
-    @BeforeEach
-    void setUp() {
-        try {
-            Results ingestionResourcesResult = generateIngestionResourcesResult();
-            Results ingestionAuthTokenResult = generateIngestionAuthTokenResult();
+    @BeforeAll
+    static void setUp() throws DataClientException, DataServiceException {
+        when(clientMock.execute(Commands.INGESTION_RESOURCES_SHOW_COMMAND))
+                .thenReturn(generateIngestionResourcesResult());
 
-            when(clientMock.execute(Commands.INGESTION_RESOURCES_SHOW_COMMAND))
-                    .thenReturn(ingestionResourcesResult);
+        when(clientMock.execute(Commands.IDENTITY_GET_COMMAND))
+                .thenReturn(generateIngestionAuthTokenResult());
 
-            when(clientMock.execute(Commands.IDENTITY_GET_COMMAND))
-                    .thenReturn(ingestionAuthTokenResult);
-
-            resourceManager = new ResourceManager(clientMock);
-        } catch (Exception e) {
-            e.printStackTrace();
-        }
+        resourceManager = new ResourceManager(clientMock);
     }
 
-    @AfterEach
-    void tearDown() {
+    @Test
+    void getIdentityTokenReturnsCorrectToken() throws IngestionServiceException, IngestionClientException {
+        assertEquals(AUTH_TOKEN, resourceManager.getIdentityToken());
     }
 
     @Test
-    void getIdentityToken() {
-        try {
-            assertEquals(AUTH_TOKEN, resourceManager.getIdentityToken());
-        } catch (Exception e) {
-            e.printStackTrace();
+    void getStorageUriVerifyRoundRubin() throws IngestionServiceException, IngestionClientException {
+        String storage;
+        HashMap<String, Integer> m = new HashMap();
+
+        for (int i = 0; i < 10; i++) {
+            storage = resourceManager.getIngestionResource(ResourceManager.ResourceType.TEMP_STORAGE);
+            m.put(storage, m.getOrDefault(storage, 0) + 1);
         }
+
+        assertEquals(5, (int) m.get(STORAGE_2));
```

Comment:
Not necessarily verifies round robin, only equal number of times each storage was used.